autoload/ale/sign.vim

```diff
@@ -67,10 +67,14 @@ endfunction
 " Given a loclist, combine the loclist into a list of signs such that only
 " one sign appears per line. Error lines will take precedence.
 " The loclist will have been previously sorted.
-function! ale#sign#CombineSigns(loclist) abort
+function! ale#sign#CombineSigns(buffer, loclist) abort
     let l:signlist = []
 
     for l:obj in a:loclist
+        if has_key(l:obj, 'bufnr') && l:obj.bufnr >= 0 && l:obj.bufnr != a:buffer
```

Comment:
Removing these warnings/errors is harmful for this linter. Your package might not compile, yet there may be no indication why.

What about adding a new flag like `g:ale_experimental_multibuffer` that is strictly opt-in? If it is true, then multibuffer would work as it does presently in this diff. If not, this linter will strip errors/warnings for other buffers.

This would give you a good, nonintrusive way to start testing the feature while requiring minimal changes here.

To answer your questions:

> Will it break echo line as cursor moves?

No. Tested.

> What effect will it have on the warning/error count?

Counts are the total number of errors/warnings that the linter returned. This seems correct in my view as my statusline will show that an error or warning exists.

> Will it break the commands for jumping between warnings/errors?

I use [unimpaired](https://github.com/tpope/vim-unimpaired) for jumping around using `]l` and `[l` for example. Those work correctly. Is there another way I can test?